pycrires/pipeline.py

```diff
@@ -3884,6 +3884,140 @@ def correct_wavelengths(
+            tw_file = self.product_folder / f"cr2res_obs_nodding_trace_wave_{nod_ab}_{count_exp:03d}.fits"
+            sof_open.write(f"{tw_file} UTIL_WAVE_TW\n")
+            sof_open.close()
+
+            # Get data on size of the slit on the detector
+            tw_data = fits.open(tw_file)[1].data
+            slit_range = tw_data['SlitFraction'][0]
+            num_orders = tw_data['All'].shape[0]
+            tot_slit_fraction = slit_range[-1] - slit_range[0]
+            full_slit_length = 10 #arcsec
+            
+            # If extraction_length not specified, calculate the pixel scale
+            if extraction_length == None:
+                slit_fraction_pixels = np.mean(tw_data['Upper']-tw_data['Lower'])
+                pixel_scale = full_slit_length * tot_slit_fraction  / slit_fraction_pixels
+                extraction_length = pixel_scale
+
+            # Calculate the spatial size for the extraction and
+            # the number of spatial points
+            extraction_fraction = extraction_length / full_slit_length
+            N_points = spatial_oversampling * tot_slit_fraction / extraction_fraction
+
+            # Round N_points up to the next odd integer
+            N_points = int(np.ceil(N_points) // 2 * 2 + 1)
+            
+
+            # Create EsoRex configuration file if not found
+            self._create_config("cr2res_util_extract", "util_extract", verbose)
+            config_file = self.config_folder / "util_extract.rc"
+
+            # Calculate slit fraction centers to use for util_extract
+            extraction_centers = np.linspace(0,
+                                             tot_slit_fraction - extraction_fraction,
+                                             N_points)
+            # Make sure the middle value is centered around the star
+            extraction_centers = (extraction_centers - np.median(extraction_centers)) + slit_range[1]
+            
+            # Allocate arrays to save the results
+            num_det = 3
+            results_2d = np.zeros((num_det, num_orders, N_points, 2048))
+            errors_2d = np.zeros((num_det, num_orders, N_points, 2048))
+            wavelengths = np.zeros((num_det, num_orders, 2048))
+
+            for i, center in enumerate(extraction_centers):
+                lower_lim = center - extraction_fraction/2
+                upper_lim = center + extraction_fraction/2
+                print(f'--> Extracting spectrum between slit fractions: '
+                      f'{lower_lim:.3f} - {upper_lim:.3f}')
+                
+                # Run EsoRex
+                esorex = [
+                    "esorex",
+                    f"--recipe-config={config_file}",
+                    f"--output-dir={self.product_folder}",
+                    "cr2res_util_extract",
+                    f"--slit_frac={lower_lim},{upper_lim}",
+                    f"--oversample={util_extract_oversample}",
```

Comment:
I would prefer not to overwrite esorex parameters from the command line, unless needed. I would therefore remove the ``util_extract_oversample`` parameter from this method and instead overwrite the default value with 12 in the configuration file. Then, if the user wants to change the value, that will be possible by adjusting the configuration file. Have a look at `_create_config` with the `obs_nodding` recipe where I had also change the oversampling to 12.

Reply:
I think I have done this now, but please check if this is indeed the way you want it

Reply:
Thanks, yes that looks good!